AM.md

```diff
@@ -0,0 +1,203 @@
+
+## The Key Specifics in Scripting in AM
+
+AM offers an interface in its console for
+
+### Example
+
+In the following example, we will extend an authentication procedure used by AM to verify user identity.
+
+1. The Authentication Chain Example
+
+    1. The Administrative Interface
+
+        To sign in with administrative configuration, navigate to your AM instance `/console` path. For example:
+
+        https://default.iam.example.com/am/console/
+
+        > AM allows to use separate authentication chains for administrative and non-administrative users. The choice can be made in AM's administrative console under Realms > Top Level Realm > Authentication > Settings > Core—by selecting desired values for Administrator Authentication Configuration and Organization Authentication Configuration inputs.
+        If you don't use the `/console` path, the default _Organization Authentication Configuration_ chain will be used, regardless of whether the actual user is an administrator or not.
+        >
+        > This may not matter if both options point to the same authentication chain.
+
+    1. The Client-Side Script
+
+        1. Navigate to Top Level Realm > Scripts.
+
+            You will see number of predefined scripts, some of which can serve as templates for the new ones.
+
+        1. Select + New Script.
+
+            In the New Script dialog, provide Name and select Client-side Authentication for the Script Type input. For example:
+
+            * Name:  Scripted Module - Client Side Example
+            * Script Type: Client-side Authentication
+
+        1. Select the Create button.
+
+            In the next dialog, with the new script properties, populate the Script input with the following JavaScript code:
+
+            * Script:
+
+            ```javascript
+            /*
+            * Object to hold the data to be passed to the server-side processing.
+            *
+            * The name of this object can be any valid variable name.
+            */
+            var data = {};
+
+            /*
+            * Script element for loading an external library.
+            *
+            * When the script is loaded, it will make a request to an external source to obtain the client's IP information.
+            *
+            * The information, received as a JSON object, is then saved as a string in the
+            * value of the `output` input in the form provided on the client side automatically. The value will become available to the server-side script as the `clientScriptOutputData` javascript variable.
+            */
+            var script = document.createElement('script');
+
+            script.src = 'https://code.jquery.com/jquery-3.4.1.min.js';
+            script.onload = function (e) {
+                $.getJSON('https://ipgeolocation.com/?json=1', function (json) {
+                clientScriptOutputData.ip = json;
+                    output.value = JSON.stringify(data);
```

Comment:
Is `output` correct here? It appears your above variable is called `data`. Also, where is `submit` defined?

Reply:
The automatically generated page contains a form with the `output` input and some JS including `submit()`.

Reply:
But I think you may be looking at an older commit (that has captured incorrect state). The script is supposed to look like this:

            ```javascript
            var script = document.createElement('script'); // 1

            script.src = 'https://code.jquery.com/jquery-3.4.1.min.js';
            script.onload = function (e) { // 2
                $.getJSON('https://ipgeolocation.com/?json=1', function (json) {
                    output.value = JSON.stringify({
                        ip: json
                    }); // 3

                    submit(); // 4
                });
            }

            document.getElementsByTagName('head')[0].appendChild(script); // 1

            autoSubmitDelay = 4000; // 5
            ```